synapse/storage/databases/main/metrics.py

```diff
@@ -282,8 +282,8 @@ def _generate_user_daily_visits(txn):
             now = self._clock.time_msec()
 
             sql = """
-                INSERT INTO user_daily_visits (user_id, device_id, timestamp)
-                    SELECT u.user_id, u.device_id, ?
+                INSERT INTO user_daily_visits (user_id, device_id, timestamp, user_agent)
+                    SELECT u.user_id, u.device_id, ?, u.user_agent
```

Comment:
This isn't in the `GROUP BY` clause, and in general there may technically be multiple user agents associated with a device so we'll need to decide which one to record. In an ideal world I'd probably say we should do one of:

1. record every UA and make the useragent column an array (but that doesn't work on sqlite, and may make processing harder);
2. record the UA that we saw the most, but that probably involves complicated SQL; or
3. take the most recent UA we saw (easy on postgres, hard on sqlite).

All of which sound like faff, so assuming that we don't really care too much which one we pick I'd probably do `MAX(u.user_agent)`. 

(Note, this query probably works fine on SQLite as it allows adding columns not in the GROUP BY clause, it'll return an entry from a random row, but postgres refuses to allow it).